design/authenticators/authn_gcp/authn_gcp_solution_design.md

```diff
@@ -0,0 +1,576 @@
+| `authn-gcp/service_account_id`                                       | `sub` claim                            |
+| `authn-gcp/service_account_email`                                    | `email` field                          |
+
+### Access GCP authenticator
+Send the following POST request:
+
+`https://<DAP-server-hostname>/authn-gcp/mygcp/<account>/myapp/authenticate`
+
+|  | |
+|----------|-----------------|
+| **Header**          |  Content-Type: application/x-www-form-urlencoded               |
+| **Body**         |  The body must include the GCP jwt token for GCE instance. jwt=eyJhbGciOiJSUzI1NiIs......uTonCA               |
+
+### Design
+When we got the requirement to implement GCP authenticator the first thought was: if we had a generic JWT authenticator we could support it, 
+maybe not we the best UX but sure we could present a proof of concept.
+
+So the goal of this deign is to present a GCP implementation plan with a small effort to support JWT authenticator as well,
+While the main differences between them is UX and configuration options:
+GCP authenticator - should be focus on UX in google platform
+JWT authenticator - should be generic as possible to support any Vendor in the future 
+
+#### Class diagram
+The following table represents all the configuration options i want to have in JWT and its comparison to GCP
+
+|                                                                    |     JWT                                   |     GCP                                                                             |
+|--------------------------------------------------------------------|-------------------------------------------|-------------------------------------------------------------------------------------|
+|     api                                                            |     authn-jwt/                            |     authn-gcp/                                                                      |
+|     Mandatory base claims to validate                              |     Iss, exp, nbf (Configurable)          |     iss, exp, aud, iat (Constants)                                           |
+|     Mandatory base claims to validate                              |     Iss, exp, nbf (Configurable)          |     iss, exp, aud, iat, nbf (Constants)                                           |
+|     Optional base claims to validate (if exists in the token)      |     aud,sub,iat (Configurable)            |     -                                                                               |
+|     provider-uri                                                   |     Configurable   (list?)                |     Constant                                                                        |
+|     Public-certificate                                             |     Configurable (In case of self sign)   |     -                                                                        |
```

Comment:
List??
If we're accepting multiple URIs we should accept self-signed certificate for each one of them...
Actually I think that `provider-uri` should not be a list.

Reply:
me too :)